### PR TITLE
Abduction Update 2

### PR DESCRIPTION
**Pull request by AnturK**
#### New Features
- Advanced Camera now has all the abductor console functionality and more
- Pinpoint teleportation takes 8 seconds and spawns a flashy silhouette at the target location. Can be used as distraction though since it appears even if teleporter is empty.(Totally not unintended feature)
- Old area teleport left as emergency in case cameras get turned off.
- Three new negative glands:
- **Bloody gland** : sprays blood on everything in sight and deals medium brute damage to hot
- **Bodysnatch** gland : Turns host into a cocoon from with hatches snpc looking like the host - soon with equipment
- **Plasma gland** : Explodes the host in a cloud of plasma
- No-Track Helmet for the agent
#### Minor Changes
- Abductors are now virus immune (Skeletons/Golems too)
- Dissection surgery ignores clothes now
- Non-abductors can now escape the ships by mashing consoles for 10 seconds
#### Fixes & Refactors
- Split machinery into separate files
- Gland first activation is not instant anymore.
- Attacking console with abductor equipment now links them (for easier adminnery and such)

More to come as i recover/write again stuff

### CODE REVIEW

**Comment by Alek2ander** [comment]
Things abductors need:
Medical healy stuff.
Spare superbatons because the agent is likely to be disarmed at some point.
HUD icons so the scientist can see the agent in a disguise.
Jump to agent button for the camera console?

**Comment by Miauw62** [comment]
I feel like there should be a non-ayylien advanced camera console available, for when admins want one.

**Comment by AnturK** [comment]
There is,  this is just for abuctor subtype

**Comment by RemieRichards** [comment]
@Miauw62 
```/obj/machinery/computer/camera_advanced/abductor``` is what's introduced in this PR.
the ```/obj/machinery/computer/camera_advanced``` type has the main functionality (AI view), and is already live on the servers.

**Comment by Miauw62** [comment]
oh, nice.
(im too lazy to actually look at the code).

there is also a bug where advanced camera consoles give you the static image in areas without camera coverage, even if you're not actually standing at the camera console. pops up seemingly randomly.

File: code/game/gamemodes/abduction/machinery/console.dm [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+			TeleporterRetrieve()
+		else if(href_list["flip_vest"])
+			FlipVest()
+		else if(href_list["select_disguise"])
+			SelectDisguise()
+		src.updateUsrDialog()
+
+/obj/machinery/abductor/console/proc/TeleporterSet()
+	var/A = null
+	A = input("Select area to teleport to", "Teleport", A) in teleportlocs
```
Comment (Giacom) [inline code/game/gamemodes/abduction/machinery/console.dm]: After every input you need to recheck if they are able to still perform this action, as they can move away from the machine before choosing an answer.

File: code/game/gamemodes/abduction/machinery/console.dm [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	var/datum/browser/popup = new(user, "computer", "Abductor Console", 400, 500)
+	popup.set_content(dat)
+	popup.set_title_image(user.browse_rsc_icon(src.icon, src.icon_state))
+	popup.open()
+	return
+
+/obj/machinery/abductor/console/Topic(href, href_list)
+	if(..())
+		return
+	if((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
```
Comment (Giacom) [inline code/game/gamemodes/abduction/machinery/console.dm]: Most of these checks are not required as the code above

```
if(..())
    return
```

contains the checks. ..(); to elaborate, ..() calls /obj/machinery/Topic() which itself contains the checks which will return 1 if they are invalid and return 0 if they are valid. (It has always been like that, doesn't make much sense)

File: code/game/gamemodes/abduction/machinery/dispenser.dm [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+		user.drop_item()
+		W.loc = src
+		for(var/i=1,i<=gland_colors.len,i++)
+			if(gland_types[i] == W.type)
+				amounts[i]++
+
+/obj/machinery/abductor/gland_dispenser/Topic(href, href_list)
+	if(..())
+		return
+	if((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
```
Comment (Giacom) [inline code/game/gamemodes/abduction/machinery/dispenser.dm]: Same thing with the ..() checks.

**Comment by Kromgar** [comment]
For the love of god fix the race changing one killing you

**Comment by AnturK** [comment]
Glorious sprites by Ausops included now!

**Comment by RemieRichards** [comment]
Still WIP or?

**Comment by AnturK** [comment]
This batch should be good to go.

**Comment by AnturK** [comment]
Should be good now.

**Comment by RemieRichards** [comment]
@AnturK I recommend unEquipping all the gear the SNPC starts with, and deleting the gear too.

**Comment by AnturK** [comment]
The greytide ones spawn naked, not sure thats intended but works for this

**Comment by RemieRichards** [comment]
well that's weird (and I've used the greytide ones in the past, on basil, and they weren't naked :/) but ok.

**Comment by astralenigma** [comment]
Shouldn't this have a Ayy LMAO tag?

**Comment by peoplearestrange** [comment]
Be nice if there was a "Create Ayliem" transform from the Player Panel. Making it easier for admins to create them from scratch.

Maybe even a Alien "select equipment" profile too would be nice.

**Comment by astralenigma** [comment]
I mean I thought the tags meant they belonged to abduction related.

**Pull request merged**